### PR TITLE
add: custom quote block

### PR DESCRIPTION
**Pull request by CaCaBlocker**
Because:

- specific small component(props: content, source) to add on documentation.

This commit:

- add this component with props(content, source) on docs files(mdx).

### CODE REVIEW

**Comment by EiffelFly** [comment]
HI @CaCaBlocker Thanks for your first PR in instill-ai's official website, we really appreciate your contribution, it means a lot to us. I would like to bring up a suggestion about the content of the PR first.

Could you please add the structure into the PR content like below

```
Because: (You could write the reason and reference the issue here)

This commit: (Elaborate how you implement this PR)
```

You could find a example here (please don't use #close command in your PR content)

**Comment by CaCaBlocker** [comment]
Hi, @EiffelFly.
I just updated the PR content according to your request.
**Please check it and I hope your feedback.**

**Review by EiffelFly — CHANGES_REQUESTED** [review]
I think your code style is good! Please adjust these minor changes first. After that we can talk about the visual style

**Comment by CaCaBlocker** [comment]
@EiffelFly I updated the code. Please check it.

File: src/components/ui/MdxQuoteBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+export const MdxQuoteBlock = ({
+  content,
+  source,
+  position,
+}: MdxQuoteBlockProps) => {
+  return (
+    <MdxComponentBase position={position}>
+      <div className="border-l-4 border-black bg-opacity-10 px-2">
+        <p className="m-0 p-1">{content}</p>
```
Comment (EiffelFly) [inline src/components/ui/MdxQuoteBlock.tsx:18]: I think the `m-1` is redundant here, please remove it.

Comment (CaCaBlocker) [reply src/components/ui/MdxQuoteBlock.tsx]: Sorry about that. I just finished.

**Comment by EiffelFly** [comment]
@CaCaBlocker 

I had altered the style to fit our design guide better. Besides that, I had altered the structure of the component too

- I think we should not provide position prop for user to change the position of quote block. In the end the width of this component will be as same as its container (Sorry for not bringing this info upfront)
- I move the block class because it is not necessary too
- Generally, we don't recommend the nested padding in near element so I recommend removing the padding of child and move it to its parent. (This way we could make debug style easier)
- I had adjusted the dark mode variant.
- I had changed the color to adapt instill-ai color set.

Here is the component that I come up with.

```ts
export const MdxQuoteBlock = ({ content, source }: MdxQuoteBlockProps) => {
  return (
    <div className="border-l-4 border-instillGrey90 px-4 py-1 dark:border-instillGrey50">
      <div className="font-sans text-lg">{content}</div>
      <div className="text-right font-sans text-lg">
        <span>&mdash; From </span>
        <span className="font-sans text-lg underline underline-offset-2">
          {source}
        </span>
      </div>
    </div>
  );
};
```

I think after this issue we could merge this PR!!

**Comment by CaCaBlocker** [comment]
@EiffelFly 
I updated the code according to your recommendations. What you suggested are all right for me, and I didn't pay attention to such problems(for example, dark mode). I learned a lot from you.

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -31,6 +32,13 @@ type AppPropsWithLayout = AppProps & {
 
 const components = {
   CH,
+  QuoteBlock: (props: any) => (
+    <MdxQuoteBlock
+      content={props.content}
+      source={props.source}
+      position={props.position}
```
Comment (EiffelFly) [inline src/pages/_app.tsx:39]: @CaCaBlocker This position props should be removed!

**Comment by EiffelFly** [comment]
> I updated the code according to your recommendations. What you suggested are all right for me, and I didn't pay attention to such problems(for example, dark mode). I learned a lot from you.

No problem! We are really appreciate your contribution and this PR is going to be our first merged PR into frontend codebase. @CaCaBlocker Good job!

File: src/pages/_app.tsx (hunk 2 above)
Comment (CaCaBlocker) [reply src/pages/_app.tsx:39]: @EiffelFly  Sorry about that. I consider why I missed that and I'd like to hear your opinion about this update.

src/pages/_app.tsx

>   MdxYoutubeEmbed,
>   MdxQuoteBlock,
>   **MdxQuoteBlockProps,**
> } from "@/components/ui";

> QuoteBlock: (props: **MdxQuoteBlockProps**) => (
> ),

**Comment by EiffelFly** [comment]
@CaCaBlocker You may need to rebase our main again. During the development of this PR we had rolled out lots of things. Please rebase it and make sure lint is passing

**Pull request closed without merging**